lib/fbp_advanced.dart

```diff
@@ -243,26 +243,35 @@ class AdvancedFireBehaviourPredictionFormState
         CBH: _cbh,
         CFL: _cfl,
         HR: _minutes / 60.0);
-    // print('advanced: ${input}');
-    FireBehaviourPredictionPrimary prediction = FBPcalc(input, output: "ALL");
-    _onFMCChanged(prediction.FMC);
-    // Wind direction correction:
-    prediction.RAZ -= 180;
-    prediction.RAZ = prediction.RAZ < 0 ? prediction.RAZ + 360 : prediction.RAZ;
-    if (prediction.secondary != null) {
-      fireSize = getFireSize(
-          _fuelTypePreset!.code.name,
-          prediction.ROS,
-          prediction.secondary!.BROS,
-          _minutes,
-          prediction.CFB,
-          prediction.secondary!.LB);
+
+    FireBehaviourPredictionPrimary? prediction;
+    num surfaceFlameLength = 0;
+    int intensityClass = 1;
+    String? errorMessage;
+
+    try {
+      prediction = FBPcalc(input, output: "ALL");
+      // Wind direction correction:
```

Comment:
What's the intent of the correction? According to the referenced paper, RAZ is the "Spread direction azimuth" not wind direction as stated in the comment.

Reply:
(disclaimer - based on my understanding): The direction of spread - regardless of being caused by slope+aspect or wind, is often referred to as the "net effective wind direction". e.g. if the wind direction is 0, but there's a slope then fire would spread up the slope, and an "effective wind speed" calculated based on that slope. So spread direction / wind direction are often interchangeable.

Reply:
Agreed though - the terms are different. I'll update comments and be explicit about deriving an effective wind direction from the spread direction.